src/demo/demo.js

```diff
@@ -34,8 +34,66 @@ const SplatLink = withRouter(props => {
   }
   return <Link {...newProps}>{props.children}</Link>;
 });
+class EventsListener extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      gotStartLoadingEvent: false,
+      gotComponentReady: false,
+      gotComponentWillUnmount: false
+    };
+  }
+
+  componentWillMount() {
+    if (this.unSubscribeStartLoading) {
+      this.unSubscribeStartLoading();
```

Comment:
those `if`'s are never reached I believe, no?
I mean you call `this.unSubscribeStartLoading.remove()` in unmount and actually `this.unSubscribeStartLoading` is not even a function